### ISSUE TITLE
Uninitialized object doesn't have |undefined type inside arrow function callback

### ISSUE THREAD

**Issue by berickson1** [Microsoft/TypeScript#22468]
With all the strict flags turned on, I'd expect to be able to prevent an uninitialized variable from being referenced in a scoped arrow function callback.

**TypeScript Version:**  2.7.0-dev.201xxxxx

**Search Terms:** strict undefined uninitialized

**Code**

```ts
function test() {
    let foo: { bar: number };
    // Compiler error as expected
    //foo.bar = 4;
    setTimeout(() => {
        // Expect compiler error since foo has never been assigned
        console.log(foo.bar);
    }, 0);
}
test();
```
See https://github.com/berickson1/Playground/blob/master/MissingUndefined.ts

**Expected behavior:**
console.log should throw a compiler error since foo can be undefined 

**Actual behavior:**
Compiles successfully

**Playground Link:**
https://www.typescriptlang.org/play/#src=function%20test()%20%7B%0D%0A%20%20%20%20let%20foo%3A%20%7B%20bar%3A%20number%20%7D%3B%0D%0A%20%20%20%20%2F%2F%20Compiler%20error%20as%20expected%0D%0A%20%20%20%20%2F%2Ffoo.bar%20%3D%204%3B%0D%0A%20%20%20%20setTimeout(()%20%3D%3E%20%7B%0D%0A%20%20%20%20%20%20%20%20%2F%2F%20Expect%20compiler%20error%20since%20foo%20has%20never%20been%20assigned%0D%0A%20%20%20%20%20%20%20%20console.log(foo.bar)%3B%0D%0A%20%20%20%20%7D%2C%200)%3B%0D%0A%7D%0D%0A%0D%0Atest()%3B
